content/neuvector-prime/aws/contents.lr

```diff
@@ -0,0 +1,164 @@
+title: NeuVector Prime in AWS: Usage Instructions
+---
+theme: suse
+---
+body:
+
+## Prerequisites
+
+1. These instructions assume you have an EKS cluster installed and that you have access to that cluster via a `kubeconfig`.
+2. Installation requires you have the following tools available and properly configured to access your AWS account, and your EKS cluster:
+* `aws`
+* `curl`
+* `eksctl`
+* `helm` (v3 or greater)
+
+## Preparing your cluster
+
+### OIDC provider
+
+Your EKS cluster is required to have an OIDC provider installed. To check for an OIDC provider first find the OIDC issuer with the following command. Substitute `$CLUSTER_NAME` with the *Name* of the EKS cluster:
+
+```
+aws eks describe-cluster --name $CLUSTER_NAME --query cluster.identity.oidc.issuer --output text
+```
+
+A URL is returned, like `https://oidc.eks.region.amazonaws.com/id/1234567890ABCDEF`. The part after `https://` will be referred to in later instructions as the *OIDC Provider Identity*. The final section of the URL, `1234567890ABCDEF`, is the $OIDC_ID.
+
+Using the id of the issuer found above you can check if a provider is installed with the following command:
+
+```
+aws iam list-open-id-connect-providers | grep $OIDC_ID
+```
+
+If there is no output, you will need to create an OIDC provider:
+
+```
+eksctl utils associate-iam-oidc-provider --cluster $CLUSTER_NAME --approve
+```
+
+This will return the OIDC provider's ARN, with the ID at the end.
```

Comment:
If I remember correctly from Lasso conversation setting up an OIDC provider can take a rather long amount of time. Is this tsill correct? If yes we should probably give the reader a heads up that this operation takes some time.

Reply:
OIDC setup seems to only take a long time at cluster creation; it only takes a few seconds on an established cluster.